src/components/App.jsx

```diff
@@ -71,18 +72,22 @@ class App extends React.Component {
   };
 
   getIssues = async event => {
-    event.preventDefault();
-    this.setState({ isEmpty: true, isFetching: true });
-    const finalUrl = this.createUrl();
-    await fetch(finalUrl)
-      .then(this.handleErrors)
-      .then(resJson => {
-        this.setState({ isEmpty: false, isFetching: false, results: resJson, url: finalUrl });
-      })
-      .catch(err => {
-        console.error('error:', err);
-        this.setState({ fetchError: true, isFetching: false });
-      });
+      event.preventDefault();
+
+    if (!this.state.isFetching && !this.state.buttonLock) {
+      this.setState({ isEmpty: true, isFetching: true });
+      const finalUrl = this.createUrl();
+      await fetch(finalUrl)
+        .then(this.handleErrors)
+        .then(resJson => {
+          this.setState({ isEmpty: false, isFetching: false, buttonLock: true, results: resJson, url: finalUrl });
+          setTimeout(() => this.setState({ buttonLock: false }), 5000);
```

Comment:
Could we put the `setTimeout` as a callback to `setState`? Something like

```
this.setState({ isEmpty: false, isFetching: false, buttonLock: true, results: resJson, url: finalUrl }), () => {
      setTimeout(() => {
        this.setState({ buttonLock: false })
      }, 5000);
    });
  }
```

Reply:
Yeah, certainly, forgot about that little feature